src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -171,10 +171,21 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                         _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
                     };
 
-                    block.Add(F.ExpressionStatement(
-                        F.Call(receiver: builder,
-                            F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendObject),
-                            F.Convert(F.SpecialType(SpecialType.System_Object), value))));
+                    Debug.Assert(value.Type is not null);
+                    if (value.Type.IsValueType)
```

Comment:
does this work for thigns like `where T : struct` type parameters?

Reply:
it does. I'll add an explicit test

https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs#L564-L575